### PR TITLE
Added JSdocs comment labels for improved code documentation

### PR DESCRIPTION
**Pull request by NyashaNziramasanga**
Using JSdocs labels for improved code documentation when hovering on function.

For more info [jsdoc cheetsheet](https://devhints.io/jsdoc) or [jsdoc.app](https://jsdoc.app/tags-type.html)

## Before
Screen Shot 2020-07-20 at 12 33 20 am

## After
Screen Shot 2020-07-20 at 12 35 50 am

### CODE REVIEW

File: src/controllers/api.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ import { UserDocument } from "../models/User";
 
 
 /**
- * GET /api
- * List of API examples.
+ * @route   GET /api
+ * @desc    List of API examples.
```
Comment (peterblazejewicz) [inline src/controllers/api.ts:10]:
```suggestion
 * List of API examples.
 * @route GET /api
```

File: src/controllers/api.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ export const getApi = (req: Request, res: Response) => {
 };
 
 /**
- * GET /api/facebook
- * Facebook API example.
+ * @route   GET /api/facebook
+ * @desc    Facebook API example.
```
Comment (peterblazejewicz) [inline src/controllers/api.ts:20]:
```suggestion
 * Facebook API example.
 * @route GET /api/facebook
```

File: src/controllers/contact.ts [hunk 3]
```diff
@@ -11,8 +11,8 @@ const transporter = nodemailer.createTransport({
 });
 
 /**
- * GET /contact
- * Contact form page.
+ * @route   GET /contact
+ * @desc    Contact form page.
```
Comment (peterblazejewicz) [inline src/controllers/contact.ts:15]:
```suggestion
 * Contact form page.
 * @route GET /contact
```

File: src/controllers/contact.ts [hunk 4]
```diff
@@ -21,8 +21,8 @@ export const getContact = (req: Request, res: Response) => {
 };
 
 /**
- * POST /contact
- * Send a contact form via Nodemailer.
+ * @route   POST /contact
+ * @desc    Send a contact form via Nodemailer.
```
Comment (peterblazejewicz) [inline src/controllers/contact.ts:25]:
```suggestion
 * Send a contact form via Nodemailer.
 * @route POST /contact
```

File: src/controllers/home.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { Request, Response } from "express";
 
 /**
- * GET /
- * Home page.
+ * @route   GET /
+ * @desc    Home page.
```
Comment (peterblazejewicz) [inline src/controllers/home.ts:5]:
```suggestion
 * Home page.
 * @route GET /
```

File: src/controllers/user.ts [hunk 6]
```diff
@@ -10,8 +10,8 @@ import { check, sanitize, validationResult } from "express-validator";
 import "../config/passport";
 
 /**
- * GET /login
- * Login page.
+ * @route   GET /login
+ * @desc    Login page.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:14]:
```suggestion
 * Login page.
 * @route GET /login
```

File: src/controllers/user.ts [hunk 7]
```diff
@@ -23,8 +23,8 @@ export const getLogin = (req: Request, res: Response) => {
 };
 
 /**
- * POST /login
- * Sign in using email and password.
+ * @route   POST /login
+ * @desc    Sign in using email and password.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:27]:
```suggestion
 * Sign in using email and password.
 * @route POST /login
```

File: src/controllers/user.ts [hunk 8]
```diff
@@ -54,17 +54,17 @@ export const postLogin = async (req: Request, res: Response, next: NextFunction)
 };
 
 /**
- * GET /logout
- * Log out.
+ * @route   GET /logout
+ * @desc    Log out.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:58]:
```suggestion
 * Log out.
 * @route GET /logout
```

File: src/controllers/user.ts [hunk 9]
```diff
@@ -54,17 +54,17 @@ export const postLogin = async (req: Request, res: Response, next: NextFunction)
...
 export const logout = (req: Request, res: Response) => {
     req.logout();
     res.redirect("/");
 };
 
 /**
- * GET /signup
- * Signup page.
+ * @route   GET /signup
+ * @desc    Signup page.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:67]:
```suggestion
 * Signup page.
 * @route GET /signup
```

File: src/controllers/user.ts [hunk 10]
```diff
@@ -76,8 +76,8 @@ export const getSignup = (req: Request, res: Response) => {
 };
 
 /**
- * POST /signup
- * Create a new local account.
+ * @route   POST /signup
+ * @desc    Create a new local account.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:80]:
```suggestion
 * Create a new local account.
 * @route POST /signup
```

File: src/controllers/user.ts [hunk 11]
```diff
@@ -117,8 +117,8 @@ export const postSignup = async (req: Request, res: Response, next: NextFunction
 };
 
 /**
- * GET /account
- * Profile page.
+ * @route   GET /account
+ * @desc    Profile page.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:121]:
```suggestion
 * Profile page.
 * @route GET /account
```

File: src/controllers/user.ts [hunk 12]
```diff
@@ -127,8 +127,8 @@ export const getAccount = (req: Request, res: Response) => {
 };
 
 /**
- * POST /account/profile
- * Update profile information.
+ * @route   POST /account/profile
+ * @desc    Update profile information.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:131]:
```suggestion
 * Update profile information.
 * @route POST /account/profile
```

File: src/controllers/user.ts [hunk 13]
```diff
@@ -165,8 +165,8 @@ export const postUpdateProfile = async (req: Request, res: Response, next: NextF
 };
 
 /**
- * POST /account/password
- * Update current password.
+ * @route    POST /account/password
+ * @desc     Update current password.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:169]:
```suggestion
 * Update current password.
 * @route POST /account/password
```

File: src/controllers/user.ts [hunk 14]
```diff
@@ -192,8 +192,8 @@ export const postUpdatePassword = async (req: Request, res: Response, next: Next
 };
 
 /**
- * POST /account/delete
- * Delete user account.
+ * @route   POST /account/delete
+ * @desc    Delete user account.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:196]:
```suggestion
 * Delete user account.
 * @route POST /account/delete
```

File: src/controllers/user.ts [hunk 15]
```diff
@@ -206,8 +206,8 @@ export const postDeleteAccount = (req: Request, res: Response, next: NextFunctio
 };
 
 /**
- * GET /account/unlink/:provider
- * Unlink OAuth provider.
+ * @route   GET /account/unlink/:provider
+ * @desc    Unlink OAuth provider.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:210]:
```suggestion
 * Unlink OAuth provider.
 * @route GET /account/unlink/:provider
```

File: src/controllers/user.ts [hunk 16]
```diff
@@ -225,8 +225,8 @@ export const getOauthUnlink = (req: Request, res: Response, next: NextFunction)
 };
 
 /**
- * GET /reset/:token
- * Reset Password page.
+ * @route   GET /reset/:token
+ * @desc    Reset Password page.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:229]:
```suggestion
 * Reset Password page.
 * @route GET /reset/:token
```

File: src/controllers/user.ts [hunk 17]
```diff
@@ -248,8 +248,8 @@ export const getReset = (req: Request, res: Response, next: NextFunction) => {
 };
 
 /**
- * POST /reset/:token
- * Process the reset password request.
+ * @route   POST /reset/:token
+ * @desc    Process the reset password request.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:252]:
```suggestion
 * Process the reset password request.
 * @route POST /reset/:token
```

File: src/controllers/user.ts [hunk 18]
```diff
@@ -310,8 +310,8 @@ export const postReset = async (req: Request, res: Response, next: NextFunction)
 };
 
 /**
- * GET /forgot
- * Forgot Password page.
+ * @route   GET /forgot
+ * @desc    Forgot Password page.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:314]:
```suggestion
 * Forgot Password page.
 * @route GET /forgot
```

File: src/controllers/user.ts [hunk 19]
```diff
@@ -323,8 +323,8 @@ export const getForgot = (req: Request, res: Response) => {
 };
 
 /**
- * POST /forgot
- * Create a random token, then the send user an email with a reset link.
+ * @route   POST /forgot
+ * @desc    Create a random token, then the send user an email with a reset link.
```
Comment (peterblazejewicz) [inline src/controllers/user.ts:327]:
```suggestion
 * Create a random token, then the send user an email with a reset link.
 * @route POST /forgot
```

**Comment by NyashaNziramasanga** [comment]
cheers @peterblazejewicz for the review, iv committed all the changes you suggested and it's looking a lot better now

**Pull request merged**